File: game/forest/hurt_flying_falling_hang_talking.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.forest_utils import ForestUtils
from state import State


class HurtFlyingFallingHangTalking(State):
    def process_events(self, events):
        if self.get_frame_index() >= len(ForestResources.sync_catapult_hang):
            return ForestUtils.reduce_lives(self.context)
        return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.speak_catapult_hang, self.context.audio_port)
        AudioHelper.play(ForestResources.sfx_hugo_hang, self.context.audio_port)

    def render(self, screen):
        screen.blit(ForestResources.catapult_hang[12], (0, 0))
        screen.blit(Animation.get_sync_frame(ForestResources.catapult_hangspeak, ForestResources.sync_catapult_hang, self.get_frame_index()), (115, 117))

File: game-web/src/forest/hurtFlyingFallingHangTalking.ts
/** Port of game/forest/hurt_flying_falling_hang_talking.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { ForestUtils } from './forestUtils';
import type { PhoneEvents } from '../phoneEvents';

export class HurtFlyingFallingHangTalking extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    if (this.getFrameIndex() >= ForestResources.syncCatapultHang.length) {
      return ForestUtils.reduceLives(this.context, 1); // Catapult
    }
    return null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.speakCatapultHang);
    AudioHelper.play(ForestResources.sfxHugoHang);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(ForestResources.catapultHang[12], 0, 0);
    ctx.drawImage(Animation.getSyncFrame(ForestResources.catapultHangspeak, ForestResources.syncCatapultHang, this.getFrameIndex()), 115, 117);
  }
}
